smart_open/smart_open_lib.py

```diff
@@ -754,11 +754,11 @@ def HttpOpenRead(parsed_uri, mode='r', **kwargs):
 
     response = HttpReadStream(url, **kwargs)
 
-    fname = url.split('/')[-1]
+    fname = urlsplit(url, allow_fragments=False).path.split('/')[-1]
```

Comment:
May I ask about the reason for the `.split('/')[-1]`?
Would just a `if split_url.path.endswith('.gz')` check be enough?